### PR TITLE
Fix RVA field alignment

### PR DESCRIPTION
**Pull request by sbomer**
Second attempt to fix the RVA field alignment. The fix is the same as https://github.com/dotnet/cecil/pull/55, with an extra fix for the issue encountered in https://github.com/dotnet/runtime/pull/79449 and the upstream https://github.com/jbevain/cecil/pull/888. The tests are passing upstream now.

The problem was the use of `GetNextRVA` to compute the `Code` RVA, before adding that segment to the TextMap. This meant that the alignment (only added later) wasn't taken into account when writing out the code, but then later the TextMap was modified to include the alignment.

I fixed this by first inserting an aligned zero-length `Code` segment before writing the code, then inserting it again once the length is known. Removing the `Code` alignment altogether would work too, but @vitek-karas and I were thinking it might be better to keep those alignment constants in there, because it looks like they were added intentionally, even though we don't know the reason.

Note that before this change, the start of the `Code` segment (at least on x64) was not 16-byte aligned in the first place, so this change will actually move the beginning of the `Code` segment.

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]
Could you please validate that with this change:
* Trimmed HelloWorld actually runs
* It's possible to R2R - again just publish HelloWorld with trimming and R2R turned on

**Comment by vitek-karas** [comment]
Please look at backporting the fix to 7. As per our discussion, ideally we would backport just the fix, not the rest of Cecil's changes.

File: Mono.Cecil/AssemblyWriter.cs [hunk 1]
```diff
@@ -987,6 +987,11 @@ TextMap CreateTextMap ()
 			var map = new TextMap ();
 			map.AddMap (TextSegment.ImportAddressTable, module.Architecture == TargetArchitecture.I386 ? 8 : 0);
 			map.AddMap (TextSegment.CLIHeader, 0x48, 8);
+			var pe64 = module.Architecture == TargetArchitecture.AMD64 || module.Architecture == TargetArchitecture.IA64 || module.Architecture == TargetArchitecture.ARM64;
+			// Alignment of the code segment must be set before the code is written
+			// These alignment values are probably not necessary, but are being left in
+			// for now in case something requires them.
+			map.AddMap (TextSegment.Code, 0, !pe64 ? 4 : 16);
```
Comment (jkotas) [inline Mono.Cecil/AssemblyWriter.cs:994]: This does not look right to me. The AnyCPU binaries can run on 64-bit architectures and this is not going to guarantee sufficient alignment for them,

For example, consider `static ReadOnlySpan<long> MySpan => new long[] { 1, 2, 3, 6, 7 };`. The backing RVA field needs to have 8-byte alignment in this case.

Comment (jkotas) [reply Mono.Cecil/AssemblyWriter.cs:994]: Ah ok, this extra alignment applies to IL code only (and it should not be needed it in the first place). The RVA alignment is handled elsewhere.

**Pull request merged**